backend/auth.py

```diff
@@ -243,3 +249,65 @@ def handle_request(response):
 
         redirect = self.get_argument("next", '/')
         self.redirect(redirect)
+
+
+class UpdateUserHandler(handlers.SafeHandler):
+    def post(self):
+        """
+        If a user is logged in to elixir, and also has google login cookies, the
+        google users information in the database will be updated with the elixir
+        users information.
+        """
+        try:
+            # Double check so that the elixir user isn't already have any credentials
+            # in the database.
+
+            elixir_identity = self.get_secure_cookie('user')
+
+            (db.User.select()
+                    .join(db.DatasetAccess)
+                    .where(
+                           db.User.user == db.DatasetAccess.user,
+                           db.User.identity == elixir_identity)
+                    .get())
+            msg = "This elixir account already has its own credentials. Sadly, you will have to contact us directly to merge your accounts."
+            self.set_user_msg(msg, "error")
+
+            self.finish({'redirect':'/login'})
+            return
+        except db.User.DoesNotExist:
+            # This is what we want
+            pass
+
+        try:
+            # Check if we have a google login, will throw an AttributeError
+            # if the cookie isn't available
+            google_identity = self.get_secure_cookie('google_identity').decode('utf-8')
+
+            # Try to update the google user in the database with the elixir information
+            # This throws a peewee.IntegrityError if the elixir account is already in
+            # the database
+            db.User.update( name = self.get_secure_cookie('user').decode('utf-8'),
+                            email = self.get_secure_cookie('email').decode('utf-8'),
+                            identity = self.get_secure_cookie('identity').decode('utf-8'),
+                            identity_type = 'elixir'
+                            ).where( db.User.identity == google_identity ).execute()
+
+            self.set_secure_cookie('identity_type', 'updated')
+        except AttributeError:
+            # This will happen when we don't have a google cookie
+            msg = "You need to log in to a google account to be able to transfer credentials"
+            self.set_user_msg(msg, "info")
+            self.finish({'redirect':'/login'})
+            return
+        except peewee.IntegrityError:
+            # This will happen if the elixir account is already in the database
+            msg = "This elixir account is already in our database, so it can't be used to update another google account."
+            self.set_user_msg(msg, "error")
+            self.finish({'redirect':'/login'})
+            return
+
+        msg = "Your account has been updated! You may now use the site as you used to, using your Elixir account."
+        self.set_user_msg(msg, "success")
+
+        self.finish({'redirect':'/'})
```

Comment:
The final redirect should be to whatever page the user was on before. It's accessible in the backend like this:
https://github.com/NBISweden/swefreq/blob/develop/backend/auth.py#L89-L93

Maybe the previous redirect should also be to the other place, or it should be a special version of the page somehow, I'm not sure.